### ISSUE TITLE
VFS: Host root directory leak while /host_root/ disabled

### ISSUE THREAD

**Issue by AniLeo** [RPCS3/rpcs3#7950]
This doesn't have any pratical effect but it's a security issue.

How to reproduce:
- Boot /dev_flash/pspemu/psp_emulator.self directly (no argv[1] - path to psp game)

The emulator will attempt accessing Host's /host_root/, and while RPCS3 correctly returns CELL_ENOTMOUNTED, the emulator is able to access Host's full emulator path regardless. **Guest applications must not be able to peek into Host's paths while host_root is not mounted.**

Example from log:
```
E {PPU[0x1000000] Thread (main_thread) [0x002d09ac]} SYS: '_sys_prx_load_module' failed with 0x8001003a : CELL_ENOTMOUNTED, /host_root//home/ani/.config/rpcs3/dev_flash/pspemu/release/sysmem.sprx [1]
```

**Comment by elad335** [comment]
Test with #8799

**Comment by elad335** [comment]
Fixed by #10074, /dev_flash path is correctly detected now.
